src/api/apiErrors.js

```diff
@@ -0,0 +1,62 @@
+/* @flow strict-local */
+import type { ApiResponseErrorData } from './transportTypes';
+import * as logging from '../utils/logging';
+
+/**
+ * Runtime class of custom API error types.
+ */
+export class ApiError extends Error {
+  data: $ReadOnly<ApiResponseErrorData>;
+  httpStatus: number;
```

Comment:
One thing I told Boris over on #3484 I think applies: the commit that deletes `makeApiError` would be best squashed into the one that adds this, because it actually makes both commits easier to understand. In particular when reading this definition, it makes clear that choices like having those two fields with their particular names have just been moved from that other, previous code.

If you want to keep the `isEventQueueIdError` change in a separate commit (which sounds like a good idea to me), this first commit could just change `e.message === 'API'` to `e instanceof ApiError`, I think.

Reply:
If `ApiError` is going to be exposed to non-testing code anyway, I'm not sure a one-use-site trivial predicate like `isEventQueueIdError` helps at all; I might as well strip it entirely.

Which is convenient, because it's the only reason I separated out the commits in this particular way.

Reply:
Sure; sgtm!